### ISSUE TITLE
How to back propagate to all related node by only label one node in a graph

### ISSUE THREAD

**Issue by joepl-lai** [rusty1s/pytorch_geometric#1454]
## ❓ Questions & Help

Hello,
Now say i have a Graph  G which content three node A,B,C
and have edge index like [[A,B],[B,C]] , G will look like this A->B ->C.

After few graph convolution algorithm like your GAT_Conv, i specific got  A's embedding,(like data.x[0]), and put it into the downstream task in several Pytorch MLP layer,how can i use loss.backward to propagate to node B,C.
Because i only have node A's downstream task and label, and i want to update all nodes embedding in Graph G.

How can i do it !?

Thank you very much

**Comment by rusty1s** [comment]
Since node B and C are involved in the computation graph (sine you are using GNNs), intermediate node features will also receive a gradient. If you want to, in particular, receive a gradient for input node features, try setting `data.x.requires_grad = True`.

**Comment by joepl-lai** [comment]
Hello, thanks for your reply.
However, looks like it didn't work.
data.x still only have grad on target node,
Below is my simple model,
can you help me to find out what's going wrong ? Thank's a lot

class geo_gat(torch.nn.Module):
	def __init__(self,output_channel):
		super(geo_gat,self).__init__()
		self.output_channel = output_channel
		self.conv1 = GATConv(200,25,heads = 8,dropout = 0.5)
		self.conv2 = GATConv(25*8 ,200,heads = 1,concat = False,dropout = 0.5)
		self.case_mlp = Seq(Lin(200, 600), ReLU(), Lin(600, self.output_channel))
	def forward(self,x,edge_index,batch,masked,generate = False):
		x = F.elu(self.conv1(x, edge_index))
		x = F.elu(self.conv2(x, edge_index))
		case_node = x[masked]
		x.register_hook(print)
		case_node = self.case_mlp(case_node)
                return case_node
case = geo_gat(x,edge_index,masked)
case_loss = self.loss(case,y)
case_loss.backward()

**Comment by rusty1s** [comment]
There's nothing wrong. At the point in time `register_hook` gets called, you only have gradients for the nodes in `masked`. You get gradients for nodes outside of `masked` before the `conv` calls, *e.g.*:
```python
def forward(self,x,edge_index,batch,masked,generate = False):
		x = F.elu(self.conv1(x, edge_index))
		x.register_hook(print)
		x = F.elu(self.conv2(x, edge_index))
		case_node = x[masked]
		case_node = self.case_mlp(case_node)
                return case_node
```

**Comment by joepl-lai** [comment]
Thank's for your reply!
